cpp/src/parquet/geospatial_util_json_internal.cc

```diff
@@ -0,0 +1,227 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "parquet/geospatial_util_json_internal.h"
+
+#include <string>
+
+#include "arrow/config.h"
+#include "arrow/extension_type.h"
+#include "arrow/result.h"
+#include "arrow/util/string.h"
+
+#include "parquet/exception.h"
+#include "parquet/types.h"
+
+#ifdef ARROW_JSON
```

Comment:
If we mandate ARROW_JSON for Parquet, then we don't need these conditional inclusions.

Reply:
I'd rather not make it mandatory.

Reply:
Is there a specific reason? It complicates maintenance to have optional functionality like this.

Reply:
Rapidjson linked by Arrow conflicts with the one that we use internally :(

That said, I won't object to it if there is a consensus to make Rapidjson as a required dependency.

Reply:
> Rapidjson linked by Arrow conflicts with the one that we use internally :(

Then it's a bug that would deserve fixing :) Do you want to look into that?

Reply:
The Arrow-vendored RapidJSON uses the `arrow::rapidjson` namespace, so that sounds quite unexpected in any case.

Reply:
Unfortunately, these includes leak the Rapidjson headers: https://github.com/apache/arrow/blob/main/cpp/src/parquet/reader_test.cc#L32-L34

Reply:
> Then it's a bug that would deserve fixing :) Do you want to look into that?

Of course, let me fix it.

Reply:
I just double checked that there shouldn't be any issue if we link parquet with rapidjson. If removing this condition makes it easier to maintain the code, let's do it.